blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+}
+```
+
+We also needed to define the
+[IMAGE_BOOT_FILES](https://docs.yoctoproject.org/singleindex.html#term-IMAGE_BOOT_FILES)
+, which lists files that should
+be installed into the boot partition by the wic tool. We defined it in
+`kas.conf`. Note that the value of this variable heavily depends on your
+system's boot partition layout:
+
+```conf
+IMAGE_BOOT_FILES = " \
+  bootfiles/EFI/BOOT/grub*;EFI/BOOT/ \
+  bootfiles/EFI/BOOT/x86_64-efi/*;EFI/BOOT/x86_64-efi/ \
+  bzImage-initramfs-${MACHINE}.bin;bzImage-initramfs-a \
+  bzImage-initramfs-${MACHINE}.bin;bzImage-initramfs-b \
+  bzImage-initramfs-${MACHINE}.bin.sig;bzImage-initramfs-a.sig \
+  bzImage-initramfs-${MACHINE}.bin.sig;bzImage-initramfs-b.sig \
+  bootx64.efi;EFI/BOOT/ \
+"
+```
+
+At this point, we encountered several problems with our build.
+
+With `GRUB_SIGN_VERIFY` variable enabled, every grub component needed to be
+signed so that grub could use it. The recipes from `meta-efi-secure-boot`
+take
+care of generating the signatures. However, the `grubenv` signature was missing
+from our output files. As it turns out, the layer does not automatically sign
+that file. We had to append that feature to the `grub-efi` recipe:
+
+```bitbake
+# grub-efi_%.bbappend
+
+fakeroot python do_sign:append:class-target() {
+    uks_bl_sign(dir + 'grubenv', d)
+}
+
+fakeroot do_chownboot:append() {
+    chown root:root -R "${D}${EFI_BOOT_PATH}/grubenv${SB_FILE_EXT}"
+}
+
+do_deploy:append:class-target () {
+    # deploy missing grubenv.sig file
+    install -m 0600 "${D}${EFI_BOOT_PATH}/grubenv${SB_FILE_EXT}" "${DEPLOYDIR}"
+}
+```
+
+Another issue was that poky has its own `grub-efi` recipe, where it installs its
+own `grub-efi-grubx64.efi` file. It conflicted with the boot files from
+`meta-efi-secure-boot`, so we had to remove it:
+
+```bitbake
+# grub-efi_%.bbappend
+
+do_deploy:append:class-target () {
+    # remove default grub-efi-grubx64.efi file deployed by .bb from poky; when
+    # it is left in deploydir, bootimg-efi plugin picks it up after cloning
+    # files from IMAGE_BOOT_FILES list
+    # see: https://git.yoctoproject.org/poky/tree/scripts/lib/wic/plugins/source/bootimg-efi.py?id=00c04394cbc5ecaced7cc1bc8bc8787e621f987d#n360
+    rm -rf ${DEPLOYDIR}/${GRUB_IMAGE_PREFIX}${GRUB_IMAGE}
+}
+```
+
+Now your files should be signed automatically during the build. They will be
+deployed along with their signatures. By default, they will be signed using the
+sample keys from
+[meta-signing-key](https://github.com/Wind-River/meta-secure-core/tree/master/meta-signing-key/files/uefi_sb_keys)
+, which is of course extremely unsafe and should only be used for testing.
+`meta-signing-key` provides a
+[script](https://github.com/Wind-River/meta-secure-core/blob/master/meta-signing-key/scripts/create-user-key-store.sh)
+, which generates custom user keys. Such keys should be enrolled in UEFI so
+that Secure Boot can be safely utilized.
+
+## Summary
```

Comment:
Blog post does not mention Zarhus, but everything should be assumed is done as feature of Zarhus OS.

Reply:
Is this okay?
https://github.com/3mdeb/news-and-ideas/pull/590/commits/fdacbed310ba64ef2ea4554b1b24224bdb5f4c5b#diff-408ae19700697ee529e7794c3865363586a3ffe06152e4c90cb578ba42221b40R425

Reply:
Why do we say "will be"? They are already part of Zarhus since if one asks us for integration, we can deliver it without the need for additional development.

Reply:
I fixed it and also moved to to the introduction section; I think it fits better there 50978cce2d3f6125b04b7988412af360e0aea670